lib/Test2/Formatter/EventStream.pm

```diff
@@ -35,16 +36,33 @@ sub init {
 }
 
 sub write {
-    my ($self, $e, $num) = @_;
+    my ($self, $e) = @_;
 
-    $num = undef if $self->{+NO_NUMBERS};
+    my $json = $self->_event_to_json($e);
 
-    my $json = Test2::Harness::Fact->from_event($e, number => $num)->to_json;
     my $fh = $self->{+FH};
     print $fh "T2_EVENT: $json\n";
     $fh->flush;
 }
 
+{
+    my $J = JSON->new;
+    $J->indent(0);
+    $J->convert_blessed(1);
+    $J->allow_blessed(1);
+
+    sub _event_to_json {
+        my ($self, $e) = @_;
+
+        my $json = eval { $J->encode($e) };
+        my $error = $@;
+        return $json if $json;
+
+        require Data::Dumper;
+        die "JSON encoding error: $error\n" . Data::Dumper::Dumper($e);
```

Comment:
Why use Dumper ont he error instead of printing it directly? is it an object instead of a string?